src/BaseControllerV2.ts

```diff
@@ -96,21 +98,31 @@ type Json = null | boolean | number | string | Json[] | { [prop: string]: Json }
 /**
  * Controller class that provides state management, subscriptions, and state metadata
  */
-export class BaseController<S extends Record<string, unknown>> {
+export class BaseController<N extends string, S extends Record<string, unknown>> {
   private internalState: IsJsonable<S>;
 
-  private internalListeners: Set<Listener<S>> = new Set();
+  private messagingSystem: ControllerMessenger<never, { type: `${N}:state-change`; payload: [S, Patch[]] }>;
```

Comment:
I'm unsure about this event name (`state-change`). Should events use kebab-case or camel-case :thinking:

Reply:
Maybe CamelCase is best, so that it matches the controller name prefix. I figure the controller name should match the identifier we use in our codebase, to keep things simple. So that'd be camel case.

Reply:
I've updated this to use camel-case.